### PR TITLE
aws-vault - fix shell concurrency

### PR DESCRIPTION
**Pull request by alebabai**
## what
* Check if aws-vault server is already running before attempting to start

## why
* Can only run once since it attempts to bind to `169.254.169.254`

## references
* #195

### CODE REVIEW

File: rootfs/etc/profile.d/aws-vault.sh [hunk 1]
```diff
@@ -8,10 +8,14 @@ else
...
   ln -sf /localhost/.awsvault ${HOME}
-  if [ "${VAULT_SERVER_ENABLED:-true}" == "true" ]; then
+  curl -sSL -o /dev/null --stderr /dev/null http://169.254.169.254/latest/meta-data/iam/security-credentials
+  result=$?
+  if [ "${VAULT_SERVER_ENABLED:-true}" == "true" ] && [ $result -ne 0 ]; then
     echo "* Started EC2 metadata service at http://169.254.169.254/latest"
     aws-vault server &
     AWS_VAULT_ARGS+=("--server")
+  else
+    echo "* Metadata server already running"
```
Comment (osterman) [inline rootfs/etc/profile.d/aws-vault.sh]: This is not always logically correct

Comment (aknysh) [reply rootfs/etc/profile.d/aws-vault.sh]: yep, should be something like this:

if [ "${VAULT_SERVER_ENABLED:-true}" == "true" ]; then
  curl -sSL -o /dev/null --stderr /dev/null http://169.254.169.254/latest/meta-data/iam/security-credentials
  result=$?
  if [ $result -ne 0 ]; then
      echo "* Started EC2 metadata service at http://169.254.169.254/latest"
      aws-vault server &
      AWS_VAULT_ARGS+=("--server")
   else
      echo "* Metadata server already running"
   fi
fi

**Comment by osterman** [comment]
Rebase

File: rootfs/etc/profile.d/aws-vault.sh (hunk 1 above)
Comment (alebabai) [reply rootfs/etc/profile.d/aws-vault.sh]: you're right, thanks

**Pull request merged**